pages/storage.tsx

```diff
@@ -0,0 +1,270 @@
+import {
+  ChangeEvent,
+  Dispatch,
+  MouseEvent,
+  MouseEventHandler,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import styled, { css } from "styled-components";
+
+const hoverscale = (scale: number = 1.3) => css`
+  transition: transform 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
+  &:hover {
+    transform: scale(${scale});
+  }
+`;
+
+const Root = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  margin: 0;
+  padding: 0;
+
+  width: 100vw;
+  height: 100vh;
+
+  place-content: center;
+  place-items: center;
+`;
+
+type UploadFileFunction = ({
+  filename,
+  text,
+}: {
+  filename?: string;
+  text?: string;
+}) => void;
+
+type FormData = {
+  filename: string;
+  text: string;
+};
+
+const blankForm = () => ({
+  filename: "",
+  text: "",
+});
+
+const useForm = (onSubmit: UploadFileFunction) => {
+  const [data, setData] = useState(blankForm());
+
+  const submit = useCallback(
+    (d: FormData) => {
+      onSubmit(d);
+      setData(blankForm());
+    },
+    [onSubmit, setData]
+  );
+
+  return useMemo(
+    () => ({
+      data,
+      setData,
+      submit: () => submit(data),
+      onFilenameChange: (e: ChangeEvent<HTMLInputElement>) =>
+        setData((d) => ({ ...d, filename: e.target.value })),
+      onTextAreaChange: (e: ChangeEvent<HTMLTextAreaElement>) =>
+        setData((d) => ({ ...d, text: e.target.value })),
+    }),
+    [data, setData, submit]
+  );
+};
+
+const Form = ({ onSubmit, ...props }: { onSubmit?: UploadFileFunction }) => {
+  const { data, onFilenameChange, submit, onTextAreaChange } =
+    useForm(onSubmit);
+
+  return (
+    <div {...props}>
+      <h2>Type in some text to upload a file</h2>
+      <Input value={data.filename} onChange={onFilenameChange} />
+      <TextArea rows={5} value={data.text} onChange={onTextAreaChange} />
+      <Button onClick={submit}>Upload</Button>
+      <FileListing />
+    </div>
+  );
+};
+
+const StyledForm = styled(Form)`
+  width: 40em;
+
+  & > * {
+    margin: 0.5em;
+  }
+`;
+
+const FileEntryText = styled(Text)``;
+
+const FileEntryRoot = styled(Button).attrs((props) => ({
+  type: "text",
+  ...props,
+}))`
+  display: flex;
+  justify-content: space-between;
+  height: 2em;
+  flex-direction: row;
+  width: 100%;
+
+  & > * {
+    place-self: center;
+  }
+
+  ${hoverscale(1.025)}
+`;
+
+const DeleteFileButton = styled(Button)`
+  ${hoverscale()}
+`;
+
+type ModalType = {
+  enabled: boolean;
+  title: string;
+  contents: string;
+};
+
+const useModal = (filename: string = "") => {
+  const [modal, setModal] = useState({
+    enabled: false,
+    title: filename ?? "",
+    contents: "",
+  });
+
+  useEffect(() => {
+    if (!modal.title || modal.title === "") {
+      setModal((m) => ({ ...m, contents: "" }));
+      return;
+    }
+    Storage.get(modal.title, { level: "private", download: true })
+      // .then((data) => "")
+      .then((data) => data?.Body?.text?.())
```

Comment:
Inline function ,would not it be better to use ASYNC function, to avoid callback hell ?